diff --git a/includes/database/sqlite/query.inc b/includes/database/sqlite/query.inc
index 6b8a72f..68e6f14 100644
--- a/includes/database/sqlite/query.inc
+++ b/includes/database/sqlite/query.inc
@@ -71,12 +71,14 @@ class UpdateQuery_sqlite extends UpdateQuery {
    *   A database condition.
    */
   protected function removeFieldsInCondition(&$fields, QueryConditionInterface $condition) {
-    foreach ($condition->conditions() as $child_condition) {
-      if ($child_condition['field'] instanceof QueryConditionInterface) {
-        $this->removeFieldsInCondition($fields, $child_condition['field']);
-      }
-      else {
-        unset($fields[$child_condition['field']]);
+    foreach ($condition->conditions() as $key => $child_condition) {
+      if (isset($child_condition['field'])) {
+        if ($child_condition['field'] instanceof QueryConditionInterface) {
+          $this->removeFieldsInCondition($fields, $child_condition['field']);
+        }
+        else {
+          unset($fields[$child_condition['field']]);
+        }
       }
     }
   }
