diff --git a/devel_generate/file.devel_generate.inc b/devel_generate/file.devel_generate.inc
index 6577ef9..c4e76ea 100644
--- a/devel_generate/file.devel_generate.inc
+++ b/devel_generate/file.devel_generate.inc
@@ -18,8 +18,7 @@ function _file_devel_generate($object, $field, $instance, $bundle) {
       $source->uri = $path;
       $source->uid = 1; // TODO: randomize? use case specific.
       $source->filemime = 'text/plain';
-      $path_exploded = explode("//", $path);
-      $source->filename = array_pop($path_exploded);
+      $source->filename = basename($path);
       $destination_dir = $field['settings']['uri_scheme'] . '://' . $instance['settings']['file_directory'];
       file_prepare_directory($destination_dir, FILE_CREATE_DIRECTORY);
       $destination = $destination_dir . '/' . basename($path);
diff --git a/devel_generate/image.devel_generate.inc b/devel_generate/image.devel_generate.inc
index 1a12f79..c9cff37 100644
--- a/devel_generate/image.devel_generate.inc
+++ b/devel_generate/image.devel_generate.inc
@@ -33,8 +33,7 @@ function _image_devel_generate($object, $field, $instance, $bundle) {
       $source->uri = $path;
       $source->uid = 1; // TODO: randomize? Use case specific.
       $source->filemime = 'image/' . pathinfo($path, PATHINFO_EXTENSION);
-      $path_exploded = explode("//", $path);
-      $source->filename = array_pop($path_exploded);
+      $source->filename = basename($path);
       $destination_dir = $field['settings']['uri_scheme'] . '://' . $instance['settings']['file_directory'];
       file_prepare_directory($destination_dir, FILE_CREATE_DIRECTORY);
       $destination = $destination_dir . '/' . basename($path);
